vault.js

```diff
@@ -1,4 +1,21 @@
 'use strict';
 module.exports = function() {
-  
+  var obj = {};
+
+    return {
+         setValue: setValue,
+         getValue: getValue,
+    }
+
+  function setValue(key, value){
+    obj[key] = value;
+  }
+
+  function getValue(key){
+    if (!key || !obj.hasOwnProperty(key)){
```

Comment:
A slightly more concise key to check the object key is to just access through bracket notation.
`!vault[key]`